src/kernel/arch/x86/rtc.zig

```diff
@@ -206,14 +207,20 @@ fn readRtc() DateTime {
 /// The interrupt handler for the RTC.
 ///
 /// Arguments:
-///     IN ctx: *arch.InterruptContext - Pointer to the interrupt context containing the contents
+///     IN ctx: *arch.CpuState - Pointer to the interrupt context containing the contents
 ///                                      of the register at the time of the interrupt.
 ///
-fn rtcHandler(ctx: *arch.InterruptContext) void {
+fn rtcHandler(ctx: *arch.CpuState) u32 {
```

Comment:
Just for curiosity and learning's sake, why did you choose the RTC instead of the PIT for firing off task switches?

Reply:
The PIT is more accurate and so can be used as a time since boot and stuff as adding a scheduler to the handler will skew the ticks. But mostly just a design chose really, it doesn't really matter. We will probs move to the APIC soon so this may change

Reply:
Oh I see. So delaying returning from the PIT handler could cause time inaccuracies?